apps/src/code-studio/components/Libraries/LibraryCreationDialog.jsx

```diff
@@ -67,37 +104,92 @@ class LibraryCreationDialog extends React.Component {
     );
   };
 
+  validateInput = () => {
+    // Check if any of the checkboxes are checked
+    // If this changes the publishable state, update
+    let isChecked = $("input[type='checkbox']").is(':checked');
+    if (isChecked !== this.state.canPublish) {
+      this.setState({canPublish: isChecked});
+    }
+  };
+
   displayFunctions = () => {
     if (!this.state.loadingFinished) {
-      return <div>Loading...</div>;
+      return <div id="loading">Loading...</div>;
     }
+    let keyIndex = -1;
     return (
       <div>
-        <div>{this.state.libraryName}</div>
-        {this.state.selectedFunctionList.map(selectedFunction => {
-          let name = selectedFunction.functionName;
-          return <div key={name}>{name}</div>;
-        })}
+        <Heading2>
+          <b>Library Name: </b>
+          {this.state.libraryName}
+        </Heading2>
+        <form id="selectFunction" onSubmit={this.publish}>
+          <textarea
+            required
+            name="description"
+            rows="2"
+            cols="200"
+            style={styles.textarea}
+            placeholder="Write a description of your library"
+          />
+          {this.state.sourceFunctionList.map(sourceFunction => {
+            let name = sourceFunction.functionName;
+            let comment = sourceFunction.comment;
+            return (
+              <div key={keyIndex++} style={styles.functionItem}>
+                <input
+                  type="checkbox"
+                  style={styles.largerCheckbox}
+                  disabled={comment.length === 0}
+                  onClick={this.validateInput}
+                  value={keyIndex}
+                />
+                {name}
+                <br />
+                {comment.length === 0 && (
+                  <p style={styles.alert}>
+                    This function cannot be exported until you add a comment to
+                    it.
```

Comment:
i18n

Reply:
I was planning to hold off on i18n until we had a better idea if these were the final phrasings. Do you think we're there?